Extensions/mute.js

```diff
@@ -708,6 +708,40 @@ XKit.extensions.mute = new Object({
 	control_panel_div: "",
 
 	cpanel: function(mdiv) {
+		$("#xkit-mute-user").unbind("click");
+		$("#xkit-mute-user").click(() => {
+			XKit.window.show("Mute user",
+				"Enter a username to show muting options." +
+				`<input type="text" maxlength="32" placeholder="e.g. new-xkit-discussion" class="xkit-textbox" id="xkit-muting-user">`,
+				"question",
+				`<div id="xkit-mute-continue" class="xkit-button default">Mute &rarr;</div>` +
+				`<div id="xkit-close-message" class="xkit-button">Cancel</div>`);
+
+			function complain(reason) {
+				$("#xkit-muting-user")
+				.val("")
+				.attr("placeholder", reason)
+				.css("border-color", "red")
+				.click(function() {
+					$(this)
+					.unbind("click")
+					.removeAttr("style")
+					.attr("placeholder", "e.g. new-xkit-discussion");
+				});
+			}
+
+			$("#xkit-mute-continue").click(() => {
+				const username = $("#xkit-muting-user").val().trim().toLowerCase();
+
+				if (username.length == 0) {
+					complain("This can't be blank!");
+				} else if (username.includes(" ") || username.includes("_")) {
```

Comment:
This seems like it would be more comprehensive as `/^[A-Za-z0-9-]+$/.test(username)` but on the other hand regular expressions aren't the most readable things in existence so it's up to you